### DISCUSSION TITLE
PMM imputes the same values with completely different predictors

### DISCUSSION THREAD

**Discussion by matteobolner** [amices/mice#608] [Q&A] [answered]
Hello,
I have a question regarding imputation with PMM:
i want to impute missing values in a single column of a dataset; I am using PMM to impute them.
My dataset consists of around 700 samples and around 500 columns of values.
My problem is that I am seeing identical results for the imputed values regardless of the predictors that i use: 
if i use 10 columns as predictors i will get  the same imputed values as using other 10 completely different columns. 
Is this to be expected when using PMM? I would expect the results to change at least a little bit.
Thank you in advance! I hope the question is clear enough.
Matteo

**Comment by thomvolker** [comment]
Can you please provide a [reprex](https://reprex.tidyverse.org/articles/magic-reprex.html)? Without one, it is really hard to see what the issue could be. Because indeed, you would expect the imputations to change.

Thom

**Reply by matteobolner** [reply, marked as answer]
You are right, without an example it is not clear to understand.
However, I think I already found what was causing the problem.
I was trying to impute values of the column using other columns, some of which also contained missing values.
I wanted to avoid imputing the missing values in these other columns, so in the predictor matrix I set as 0 all predictors except for those in the column corresponding to my column of interest. I didn't understand correctly how to set the predictor matrix, and by doing it this way I set no predictors for my column. Therefore, even when changing the other columns, the predictors were always none. 
When setting the predictors on the row only I got NAN values for the prediction, which I now understand is to be expected when there are NANs in the predictors.
Thank you anyway! After two days of not finding anything I finally decided to post a question, and of course within 5 minutes of posting I found what I was looking for...
Hopefully this will be useful to someone else in the future.
Cheers, 
Matteo
